portalnet/src/events.rs

```diff
@@ -103,3 +104,92 @@ impl PortalnetEvents {
         }
     }
 }
+
+/// Events that can be produced by the `OverlayProtocol` event stream.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum OverlayEvent {
+    LightClientOptimisticUpdate,
+    LightClientFinalityUpdate,
+}
+
+/// Timestamp of an overlay event.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum Timestamp {
+    /// Timestamp not available.
+    NotAvailable,
+    /// Event creation time.
+    CreateTime(i64),
```

Comment:
What's the purpose of a signed number here? And what's the significance of the case of `-1` in `to_millis()`?

Reply:
Unix timestamps are in signed integers and  `-1` represents an error. It is mostly for historical reasons and deals with dates before January 1st, 1970. You can read more [here](https://unix.stackexchange.com/questions/25361/why-does-unix-store-timestamps-in-a-signed-integer) for example.

Reply:
Oh gotcha, I didn't know about the `-1` representing an error.

I would think that [this function](https://github.com/ethereum/trin/blob/73f3e1e6152de09cd56ad060aecbe3635701e94e/portalnet/src/events.rs#L167) should return a `-1` then, instead of a `0` in case of an error when getting the duration. Otherwise the signed-ness and `-1` case appear unused.

Reply:
`millis_to_epoch` makes sense to not be negative, because it is a duration from an anchor in time `UNIX_EPOCH`. 
`-1` is still used in the implementation of `From<i64>` .

I understand that we can get away with `u64`  but I think it is better to stick with the UNIX standards of timestamps which is `i64`.